core/ledger/kvledger/txmgmt/txmgr/lockbasedtxmgr/lockbased_txmgr.go

```diff
@@ -34,16 +33,16 @@ var logger = flogging.MustGetLogger("lockbasedtxmgr")
 // LockBasedTxMgr a simple implementation of interface `txmgmt.TxMgr`.
 // This implementation uses a read-write lock to prevent conflicts between transaction simulation and committing
 type LockBasedTxMgr struct {
-	ledgerid        string
-	db              privacyenabledstate.DB
-	pvtdataPurgeMgr *pvtdataPurgeMgr
-	validator       validator.Validator
-	stateListeners  []ledger.StateListener
-	ccInfoProvider  ledger.DeployedChaincodeInfoProvider
-	commitRWLock    sync.RWMutex
-	oldBlockCommit  sync.Mutex
-	current         *current
-	hasher          ledger.Hasher
+	ledgerid            string
+	db                  privacyenabledstate.DB
+	pvtdataPurgeMgr     *pvtdataPurgeMgr
+	commitBatchPreparer *validation.CommitBatchPreparer
```

Comment:
Validation and batch preparation go together. Because, the validation of a transaction T<sub>n</sub>, needs the batch that has accumulated output of all the valid transactions till the transaction T<sub>n-1</sub>. Hence, you see only one exported struct in this package - `CommitBatchPreparer`. This struct uses a `validator` to validate individual transactions and keep preparing the batch by the side.

Renaming `CommitBatchPreparer` to `Validator` would be a wrong notion.

Regarding the the package name, I am open to change it to some better name. However, `serializability` sounds like a property as oppose to a package name. I considered calling it more familiar term `mvcc` but that would technically be a wrong notion and hence left it as validation (assuming, this being inside ledger code, scope would be clear). But, as I said, I am open for a better name for this package.

Most of the remaining comments below seems like a repetition of this comment... i.e., validation v/s batch prepare.

Reply:
I see that your suggestion for package name is `txisolation`. Hmmm, not sure that I like it for the package name though.

Reply:
> I see. With that context, `CommitBatchPreparer` seems okay. As the term does convey only one operation and ignores validation, at the first look, it seemed wrong.

It does not ignore the validation. The full name is `validation.CommitBatchPreparer`. Further, the function on this is `ValidateAndPrepareBatch`